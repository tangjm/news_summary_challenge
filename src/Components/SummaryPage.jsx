import PropTypes from 'prop-types';
import { useParams } from 'react-router';

import ArticleSummaryBlock from './ArticleSummaryBlock';
import ArticleText from './ArticleText';
import ArticleModel from '../utils/ArticleModel';

const SummaryPage = ({ articleArr }) => {

	const { id } = useParams();

	const articleToDisplay = articleArr.find(currentArticle => currentArticle.id === id);
	const articleObj = articleToDisplay ? new ArticleModel(articleToDisplay.id,
		articleToDisplay.webTitle,
		articleToDisplay.webUrl,
		articleToDisplay.fields.thumbnail,
		articleToDisplay.fields.bodyText
	) : { error: "Unexpected article object" };

	return (
		<div>
			<ArticleSummaryBlock article={articleObj} />
			<ArticleText bodyText={articleObj.text ?? articleObj.error} />
		</div>
	)
}

SummaryPage.propTypes = {
	articleArr: PropTypes.arrayOf(
		PropTypes.exact({
			id: PropTypes.string,
			type: PropTypes.string,
			sectionId: PropTypes.string,
			sectionName: PropTypes.string,
			webPublicationDate: PropTypes.string,
			webTitle: PropTypes.string,
			webUrl: PropTypes.string,
			apiUrl: PropTypes.string,
			fields: PropTypes.exact({
				thumbnail: PropTypes.string,
				bodyText: PropTypes.string
			}),
			isHosted: PropTypes.bool,
			pillarId: PropTypes.string,
			pillarName: PropTypes.string
		})
	)
}

export default SummaryPage;
