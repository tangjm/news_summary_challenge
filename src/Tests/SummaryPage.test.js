import { render } from '@testing-library/react';
import SummaryPage from '../Components/SummaryPage';

test(`Article Block matches snapshot`, () => {
	const summaryPage = render(<SummaryPage />);
	expect(summaryPage).toMatchSnapshot();
})