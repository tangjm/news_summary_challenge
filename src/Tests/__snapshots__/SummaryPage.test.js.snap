// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article Block matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <img
          alt="article thumnail"
          src="https://cdn-media-1.freecodecamp.org/images/1*y6C4nSvy2Woe0m7bWEn4BA.png"
        />
        <h2>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Nostrum, quas.
        </h2>
      </div>
      <p>
        Lorem ipsum, dolor sit amet consectetur adipisicing elit. Modi, tempora natus repudiandae eum possimus asperiores eligendi expedita quas quaerat sapiente magnam labore autem aspernatur impedit in cum veritatis ipsum? Explicabo tempora in sed iure minima minus, aspernatur beatae quaerat quis quam labore optio, quibusdam blanditiis maiores molestiae fugiat rem? Id beatae optio consectetur sed, aliquam explicabo maiores error nobis consequatur facere commodi laboriosam temporibus incidunt odio magnam cupiditate quia veritatis dolorum possimus? Autem voluptates enim quae obcaecati doloremque et esse, tempora ea facere, at incidunt omnis nostrum velit possimus ipsum animi architecto sapiente veniam vitae? Consequatur excepturi repudiandae sapiente! Quod.
      </p>
    </div>
  </body>,
  "container": <div>
    <div>
      <img
        alt="article thumnail"
        src="https://cdn-media-1.freecodecamp.org/images/1*y6C4nSvy2Woe0m7bWEn4BA.png"
      />
      <h2>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Nostrum, quas.
      </h2>
    </div>
    <p>
      Lorem ipsum, dolor sit amet consectetur adipisicing elit. Modi, tempora natus repudiandae eum possimus asperiores eligendi expedita quas quaerat sapiente magnam labore autem aspernatur impedit in cum veritatis ipsum? Explicabo tempora in sed iure minima minus, aspernatur beatae quaerat quis quam labore optio, quibusdam blanditiis maiores molestiae fugiat rem? Id beatae optio consectetur sed, aliquam explicabo maiores error nobis consequatur facere commodi laboriosam temporibus incidunt odio magnam cupiditate quia veritatis dolorum possimus? Autem voluptates enim quae obcaecati doloremque et esse, tempora ea facere, at incidunt omnis nostrum velit possimus ipsum animi architecto sapiente veniam vitae? Consequatur excepturi repudiandae sapiente! Quod.
    </p>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
